sw/applications/example_spi_write/main.c

```diff
@@ -0,0 +1,176 @@
+/**
+ * @file main.c
+ * @brief Simple spi write example using BSP
+ *
+ * Simple example that writes a 1kB buffer to flash memory at a specific address
+ * and then read it back to check if the data was written correctly.
+ * 
+ * @note The application assume the correct functioning of the read operation.
+ *
+*/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "x-heep.h"
+#include "w25q128jw.h"
+
+
+// Test functions
+void test_write(uint32_t *test_buffer, uint32_t len);
+void test_write_dma(uint32_t *test_buffer, uint32_t len);
+void test_write_quad(uint32_t *test_buffer, uint32_t len);
+void test_write_quad_dma(uint32_t *test_buffer, uint32_t len);
+
+// Check function
+void check_result(uint32_t *test_buffer, uint32_t len);
+
+
+// Start buffers:
+#include "buffer.bin"
+// End buffer
+uint32_t flash_data[256];
+// Flash address to write to
+#define FLASH_ADDR 0x00008500
+// Global flag to keep track of errors
+uint32_t global_errors = 0;
+// Define global status variable
+w25q_error_codes_t global_status;
+
+int main(int argc, char *argv[]) {
+    printf("BSP write test\n\r");
+
+    /*
+     * Assign the test buffer to the buffer to write to flash.
+     * The buffer is defined in the file buffer.bin. As multiple buffers can
+     * be defined, this is userful to pick the right one.
+     * Also the length is specified, to test different length cases. In any case
+     * length <= test_buffer length. 
+    */
+    uint32_t *test_buffer = flash_original_1024B;
+    uint32_t len = 1024;
+
+    // Pick the correct spi device
+    spi_host_t spi;
+    #ifndef USE_SPI_FLASH
+    spi.base_addr = mmio_region_from_addr((uintptr_t)SPI_HOST_START_ADDRESS);
+    #else
+    spi.base_addr = mmio_region_from_addr((uintptr_t)SPI_FLASH_START_ADDRESS);
+    #endif
+
+    // Init SPI host and SPI<->Flash bridge parameters 
+    global_status = w25q128jw_init(spi);
+    if (global_status != FLASH_OK) return EXIT_FAILURE;
+
+    // Test simple write
+    test_write(test_buffer, len);
+
+    // Test simple write with DMA
+    test_write_dma(test_buffer, len);
+
+    // Test quad write
+    test_write_quad(test_buffer, len);
+
+    // Test quad write with DMA
+    test_write_quad_dma(test_buffer, len);
+
+    if (global_errors == 0) {
+        printf("All tests passed!\n\r");
+        return EXIT_SUCCESS;
+    } else {
+        printf("Some tests failed!\n\r");
+        return EXIT_FAILURE;
+    }
+}
+
+void test_write(uint32_t *test_buffer, uint32_t len) {
+    // Write to flash memory at specific address
+    global_status = w25q128jw_write_standard(FLASH_ADDR, test_buffer, len);
+    if (global_status != FLASH_OK) return EXIT_FAILURE;
```

Comment:
this is wrong, if you want to return smth then function can not be void

Reply:
Oooops